src/main/java/com/softserve/repository/impl/TeacherRepositoryImpl.java

```diff
@@ -86,4 +86,21 @@ protected boolean checkReference(Teacher teacher) {
                         " where t.userId = null ")
                 .getResultList();
     }
+
+    @Override
+    public Optional<Teacher> getExistingTeacher(Teacher teacher) {
+        return sessionFactory.getCurrentSession().createQuery(
+                "select t from Teacher t " +
+                        "where t.name = :tName and " +
```

Comment:
Do you really need to search by this set of fields? Maybe is there some unique field?

Reply:
We need to check the exact teacher we provide. Not only the id, or name.  For Example we could have teacher1(Ivan, Ivanovich, Ivanov, professor) and teacher2(Ivan, Ivanocich, Shevchenko, professor) and only 1 field differnces but 2 different people. So we need to check through all fields.